source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRLayoutHelper.mm

```diff
@@ -0,0 +1,260 @@
+            unsigned int padding = hostConfig->GetSpacing().paddingSpacing;
+            unsigned int spacing = getSpacing(elem->GetSpacing(), hostConfig);
+            long numberOfColumns = columnSetElem->GetColumns().size();
+            // removing padding of coloumSet and spacing between elemenets
+            float availableSpace = (parentWidth - (2 * padding) - ((numberOfColumns - 1)*spacing));
+            
+            float childrenWidth = availableSpace/numberOfColumns ;
+            for (const auto &column : columnSetElem->GetColumns())
+            {
+                [self distribute:childrenWidth rootView:rootView forElement:column andHostConfig:config];
+            }
+            break;
+        }
+        case CardElementType::Table:
+        {
+            std::shared_ptr<Table> table = std::dynamic_pointer_cast<Table>(elem);
+            std::vector<std::shared_ptr<TableColumnDefinition>> colums = table->GetColumns();
+            long numberOfColumns = colums.size();
+            int cellSpacing = [config getHostConfig]->GetTable().cellSpacing;
+            float availableSpace = (parentWidth - ((numberOfColumns - 1)*cellSpacing));
+           
+            //remove spacing b/w cells, padding is taken care by the cell renderer
+            float childrenWidth = availableSpace/numberOfColumns;
+            std::vector<std::shared_ptr<TableRow>> tableRows = table->GetRows();
+            for (const auto &row : tableRows) 
+            {
+                std::vector<std::shared_ptr<TableCell>> tableCells = row->GetCells();
+                for (const auto &cell : tableCells) 
+                {
+                    [self distribute:childrenWidth rootView:rootView forElement:cell andHostConfig:config];
+                }
+            }
+            
+            //set pixel width also for Table Columns
+            for (const auto &colum : colums)
+            {
+                colum->SetPixelWidth(childrenWidth);
+            }
+            
+            break;
+        }
+            
+        case CardElementType::Column:
+        {
+            std::shared_ptr<Column> column = std::dynamic_pointer_cast<Column>(elem);
+            /// set column width explicitly, this way we will bypass existing logic
+            column->SetPixelWidth(parentWidth);
+            break;
+        }
+            
+        default:
+            break;
+    }
+}
+
+- (BOOL)shouldUseNewLayout:(ACRView *)rootView
+                forElement:(std::shared_ptr<AdaptiveCard> const &)card
+             andHostConfig:(ACOHostConfig *)config
+{
+    NSObject<ACRIFeatureFlagResolver> *featureFlagResolver = [[ACRRegistration getInstance] getFeatureFlagResolver];
+    BOOL isFlowLayoutEnabled = [featureFlagResolver boolForFlag:@"isFlowLayoutEnabled"] ?: NO;
+    
+    if(!isFlowLayoutEnabled)
+    {
+        return NO;
+    }
+
+    BOOL isGridLayoutEnabled = [featureFlagResolver boolForFlag:@"isGridLayoutEnabled"] ?: NO;
+
```

Comment:
this can be combined into one check.

Reply:
Trying to provide explicit pattern for getting values from FeatureFlagResolver. This way we always provide a default value, even if FeatureFlagResolver is nil or it doesn't return a value. 
Although, in case of BOOL it may look redundant, but I would prefer this redundancy over missing default case.  This patter will make sense for other types like NSString or NSArray. 
Please see if make sense.